### PR TITLE
fixed tooltip shows up on Mobile Web while navigating the reports

### LINKED ISSUE #1490: Expensify chat - web/desktop, when hovering over a user's name anywhere, show their primary login details
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/master/CONTRIBUTING.md) for onboarding!
___

**Deliverable:** 
On web/desktop, when hovering over a user's name anywhere, show their primary login. This includes:

- In the header of a DM (whether single or group)
- In the chat itself (ie, the header of a new comment line)
- The chat row in the LHN (whether single or group)
- The chat row in the search results (whether single or group)

image

Couple of other notes:

When the full list of names is hidden behind the ... ellipsis for Group DMs in the LHN/search results, I'd suggest we show a grouped list of all the participants' primaryLogins on hover of the ellipsis
for users that have phone numbers as primaryLogins, let's drop the @expensify.sms portion, so it's displayed on hover as: +447814266906 and not +447814266906@expensify.sms

https://www.upwork.com/jobs/~0146d36b55605d9064

### PR DESCRIPTION
**Pull request by parasharrajat**
Please review @roryabraham 

### Details
Blocked the hover events on the `Hoverable` component on touch events.

### Fixed Issues
Fixes #1490
https://github.com/Expensify/Expensify.cash/pull/1632#issuecomment-805954359

### Tests
Open the App on any Mobile web platform and click on the names on the report title on the Home page. It should open the report and no tooltip should be shown.

### Tested On
- [x] Mobile Web

### Screenshots
N/A 

https://user-images.githubusercontent.com/24370807/112365932-3b55e980-8cfe-11eb-978e-c63f7c334a23.mp4

### CODE REVIEW

File: src/components/Hoverable/index.js [hunk 1]
```diff
@@ -22,10 +22,25 @@ class Hoverable extends Component {
 
     componentDidMount() {
         document.addEventListener('mousedown', this.resetHoverStateOnOutsideClick);
+
+        // we like to Block the hover on touch devices but we keep it for Hybrid devices so
+        // following logic blocks hover on touch devices.
+        this.touchStartListener = () => {
+            this.hoverDisabled = true;
+        };
+        this.touchEndListener = () => {
```
Comment (roryabraham) [inline src/components/Hoverable/index.js:31]:
```suggestion
        this.enableHover = () => {
```

File: src/components/Hoverable/index.js [hunk 2]
```diff
@@ -22,10 +22,25 @@ class Hoverable extends Component {
 
     componentDidMount() {
         document.addEventListener('mousedown', this.resetHoverStateOnOutsideClick);
+
+        // we like to Block the hover on touch devices but we keep it for Hybrid devices so
+        // following logic blocks hover on touch devices.
+        this.touchStartListener = () => {
```
Comment (roryabraham) [inline src/components/Hoverable/index.js:28]:
```suggestion
        this.disableHover = () => {
```

File: src/components/Hoverable/index.js [hunk 3]
```diff
@@ -34,9 +49,14 @@ class Hoverable extends Component {
      * @param {Boolean} isHovered - Whether or not this component is hovered.
      */
     setIsHovered(isHovered) {
-        if (isHovered !== this.state.isHovered) {
+        if (isHovered !== this.state.isHovered && !(isHovered && this.hoverDisabled)) {
             this.setState({isHovered}, isHovered ? this.props.onHoverIn : this.props.onHoverOut);
         }
+
+        // we reset the Hover block in case touchmove was not first after touctstart
```
Comment (roryabraham) [inline src/components/Hoverable/index.js:56]: Can you help me understand the race condition that can occur here?

**Review by roryabraham — CHANGES_REQUESTED** [review]
Overall looks good @parasharrajat, just some small requests on the comments and variable naming.

File: src/components/Hoverable/index.js (hunk 3 above)
Comment (parasharrajat) [reply src/components/Hoverable/index.js:56]: I think it is possible that the touchmove never fires.

File: src/components/Hoverable/index.js (hunk 2 above)
Comment (parasharrajat) [reply src/components/Hoverable/index.js:28]: I have to name two event callbacks. Can you suggest both please?

**Review by roryabraham — APPROVED** [review]
Okay, seems to work well. Thanks

**Pull request merged**